src/mbed_tools/build/config.py

```diff
@@ -42,6 +42,21 @@ def generate_config(target_name: str, toolchain: str, program: MbedProgram) -> p
     return cmake_config_file_path
 
 
+def is_image_hex(target_name: str, program: MbedProgram) -> bool:
+    """Check if a target's image format is hex instead of bin.
+
+    Args:
+        target_name: Name of the target to check image format for.
+        program: The MbedProgram to extract targets data from.
+
+    Return:
+        True if the target's "OUTPUT_EXT" is "hex" in targets.json.
+    """
+    targets_data = _load_raw_targets_data(program)
```

Comment:
Rather than loading and parsing `targets.json` again here just to find the output extension, should we refactor `generate_config` to separate the rendering and writing of the output file from the config assembly? Then we can just pull `OUTPUT_EXT` out of the `Config` object in `build`.

Reply:
Returning a `Config` is a nice idea! I was wondering how to avoid loading json twice.

Currently generate_config returns the path to the output file, I'm changing it to also return the config itself.